test_Knight.py

```diff
@@ -0,0 +1,111 @@
+import pytest
+import chess_engine
+from Piece import Knight, Rook, Pawn
+from enums import Player
+
+
+@pytest.fixture
+def tested_knight():
+    return Knight('n', 3, 4, Player.PLAYER_1)
+
+
+@pytest.fixture
+def initial_game_state():
+    game_state = chess_engine.game_state()
+    white_pawn_1 = Pawn('p', 2, 2, Player.PLAYER_2)
+    white_pawn_2 = Pawn('p', 1, 3, Player.PLAYER_2)
+    white_pawn_3 = Pawn('p', 1, 5, Player.PLAYER_2)
+    white_pawn_4 = Pawn('p', 2, 6, Player.PLAYER_2)
+    white_pawn_5 = Pawn('p', 4, 2, Player.PLAYER_2)
+    white_pawn_6 = Pawn('p', 5, 3, Player.PLAYER_2)
+    white_pawn_7 = Pawn('p', 5, 5, Player.PLAYER_2)
+    white_pawn_8 = Pawn('p', 4, 6, Player.PLAYER_2)
+    board = [
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, white_pawn_2, Player.EMPTY, white_pawn_3, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, white_pawn_1, Player.EMPTY, Player.EMPTY, Player.EMPTY, white_pawn_4,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, white_pawn_5, Player.EMPTY, Player.EMPTY, Player.EMPTY, white_pawn_8,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, white_pawn_6, Player.EMPTY, white_pawn_7, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY]
+    ]
+    game_state.board = board
+    return game_state
+
+
+def test_get_valid_piece_takes(tested_knight, initial_game_state):
+    # test_case 1 - simple knight situation: knight in the middle and can eat 8 pieces
```

Comment:
I would use documentation instead, comments make it look messy
https://swimm.io/blog/code-comments-vs-documentation/